src/test/java/uk/gov/ons/ctp/response/action/service/impl/ActionPlanJobServiceImplTest.java

```diff
@@ -232,4 +244,36 @@ public void testCreateAndExecuteActionPlanJobUnForcedExecutionPlanDoesNotRun() t
 
     Assert.assertFalse(executedJobs.size() > 0);
   }
+  
+  /**
+   * Test the endpoint forced exec method handles no open actions for an action plan gracefully
+   * @throws Exception exception thrown
+   */
+  @Test
+  public void testCreateAndExecuteActionPlanJobForcedExecutionNoActions() throws Exception {
+
+    // load fixtures
+    List<ActionPlan> actionPlans = FixtureHelper.loadClassFixtures(ActionPlan[].class);
+    List<ActionPlanJob> actionPlanJobs = FixtureHelper.loadClassFixtures(ActionPlanJob[].class);
+    List<ActionCase> actionCases = FixtureHelper.loadClassFixtures(ActionCase[].class);
+    List<Action> actions = new ArrayList<>();
+
+    // wire up mock responses
+    Mockito.when(actionPlanRepo.findOne(1)).thenReturn(actionPlans.get(0));
+    Mockito.when(actionCaseRepo.countByActionPlanFK(1)).thenReturn(new Long(actionCases.size()));
+    Mockito.when(actionRepo.countByActionPlanFKAndStateNot(1,
+        ActionDTO.ActionState.COMPLETED)).thenReturn(new Long(actions.size()));
+
+    //let it roll
+    ActionPlanJob executedJob = actionPlanJobServiceImpl.createAndExecuteActionPlanJob(actionPlanJobs.get(0));
+
+    // assert the right calls were made
+    verify(actionPlanRepo).findOne(1);
+    verify(actionCaseRepo).countByActionPlanFK(1);
+    verify(actionRepo).countByActionPlanFKAndStateNot(1, ActionDTO.ActionState.COMPLETED);
+    verify(actionPlanJobRepo, times(0)).save(actionPlanJobs.get(0));
+    verify(actionCaseRepo, times(0)).createActions(1);
+  
+    Assert.assertNull(executedJob);
```

Comment:
We probably don't need this assert not null as it would throw a NPE on line 268 if it was null